index.js

```diff
@@ -18,37 +20,31 @@ if (/index\.covered\.js$/.test(__filename)) {
 }
 
 function NYC (opts) {
-  _.extend(this, {
-    subprocessBin: path.resolve(
-      __dirname,
-      './bin/nyc.js'
-    ),
-    tempDirectory: './.nyc_output',
-    cwd: process.env.NYC_CWD || process.cwd(),
-    reporter: 'text',
-    istanbul: require('istanbul'),
-    sourceMapCache: new SourceMapCache(),
-    require: []
-  }, opts)
-
-  if (!Array.isArray(this.reporter)) this.reporter = [this.reporter]
+  opts = opts || {}
+
+  this._istanbul = opts.istanbul
+  this.subprocessBin = opts.subprocessBin || path.resolve(__dirname, './bin/nyc.js')
+  this._tempDirectory = opts.tempDirectory || './.nyc_output'
+  this._cacheDirectory = opts.cacheDirectory || './node_modules/.cache/nyc'
+  this.cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
+  this.reporter = arrify(opts.reporter || 'text')
 
   // you can specify config in the nyc stanza of package.json.
   var config = require(path.resolve(this.cwd, './package.json')).config || {}
   config = config.nyc || {}
 
   // load exclude stanza from config.
-  this.include = config.include || ['**']
-  this.include = this._prepGlobPatterns(this.include)
+  this.include = false
+  if (config.include) {
+    this.include = this._prepGlobPatterns(arrify(config.include))
+  }
 
-  this.exclude = ['**/node_modules/**'].concat(config.exclude || ['test/**', 'test{,-*}.js'])
-  if (!Array.isArray(this.exclude)) this.exclude = [this.exclude]
+  this.exclude = ['**/node_modules/**'].concat(arrify(config.exclude || ['test/**', 'test{,-*}.js']))
```

Comment:
Could call `this._prepGlobPatterns()` here rather than reassigning `this.exclude`?